File: solutions/clojure/raindrops/1/src/raindrops.clj
(ns raindrops)

(def ^:private sounds
  {3 "Pling"
   5 "Plang"
   7 "Plong"})

(defn convert [number]
  (let [sound
        (reduce (fn [raindrops [divisor sound]]
                  (if (zero? (mod number divisor))
                    (str raindrops sound)
                    raindrops))
                ""
                sounds)]
    (if (= sound "")
      (str number)
      sound)))

File: solutions/r/raindrops/1/raindrops.R
raindrops <- function(number) {
  result <- ""
  if (number %% 3 == 0) {
    result <- "Pling"
  }
  if (number %% 5 == 0) {
    result <- paste(result, "Plang", sep="")
  }
  if (number %% 7 == 0) {
    result <- paste(result, "Plong", sep="")
  }
  if (result == "") {
    toString(number)
  } else {
    result
  }
}
